source/common/http/http3/conn_pool.h

```diff
@@ -121,19 +131,26 @@ class Http3ConnPoolImpl : public FixedHttpConnPoolImpl {
   // the HTTP3 active client does.
   bool trackStreamCapacity() override { return false; }
 
+protected:
+  void onConnected(Envoy::ConnectionPool::ActiveClient&) override;
+
 private:
+  friend class Http3ConnPoolImplPeer;
+
   // Store quic helpers which can be shared between connections and must live
   // beyond the lifetime of individual connections.
   std::unique_ptr<Quic::PersistentQuicInfoImpl> quic_info_;
+  OptRef<PoolConnectResultCallback> connect_callback_;
 };
 
-ConnectionPool::InstancePtr
+std::unique_ptr<Http3ConnPoolImpl>
 allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_generator,
                  Upstream::HostConstSharedPtr host, Upstream::ResourcePriority priority,
                  const Network::ConnectionSocket::OptionsSharedPtr& options,
                  const Network::TransportSocketOptionsConstSharedPtr& transport_socket_options,
                  Upstream::ClusterConnectivityState& state, TimeSource& time_source,
-                 Quic::QuicStatNames& quic_stat_names, Stats::Scope& scope);
+                 Quic::QuicStatNames& quic_stat_names, Stats::Scope& scope,
+                 OptRef<PoolConnectResultCallback> connect_callback);
```

Comment:
are there reasons we wouldn't want this callback, or should we make it non-optional?

Reply:
Only conn pool should be interested in the connect result because it needs to track h3 status. If a cluster manager is configured with a h3 pool directly, it doesn't need such callback.